pkg/polaris/graphql/cloudcluster/cloudcluster_azure.go

```diff
@@ -0,0 +1,325 @@
+// Copyright 2025 Rubrik, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+package cloudcluster
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/google/uuid"
+	"github.com/rubrikinc/rubrik-polaris-sdk-for-go/pkg/polaris/graphql"
+	"github.com/rubrikinc/rubrik-polaris-sdk-for-go/pkg/polaris/graphql/azure"
+	"github.com/rubrikinc/rubrik-polaris-sdk-for-go/pkg/polaris/graphql/core/secret"
+)
+
+type AzureCCESSupportedInstanceType string
+
+const (
+	AzureInstanceTypeUnspecified     AzureCCESSupportedInstanceType = "TYPE_UNSPECIFIED"
+	AzureInstanceTypeStandardDS5V2   AzureCCESSupportedInstanceType = "STANDARD_DS5_V2"
+	AzureInstanceTypeStandardD16SV5  AzureCCESSupportedInstanceType = "STANDARD_D16S_V5"
+	AzureInstanceTypeStandardD8SV5   AzureCCESSupportedInstanceType = "STANDARD_D8S_V5"
+	AzureInstanceTypeStandardD32SV5  AzureCCESSupportedInstanceType = "STANDARD_D32S_V5"
+	AzureInstanceTypeStandardE16SV5  AzureCCESSupportedInstanceType = "STANDARD_E16S_V5"
+	AzureInstanceTypeStandardD8ASV5  AzureCCESSupportedInstanceType = "STANDARD_D8AS_V5"
+	AzureInstanceTypeStandardD16ASV5 AzureCCESSupportedInstanceType = "STANDARD_D16AS_V5"
+	AzureInstanceTypeStandardD32ASV5 AzureCCESSupportedInstanceType = "STANDARD_D32AS_V5"
+	AzureInstanceTypeStandardE16ASV5 AzureCCESSupportedInstanceType = "STANDARD_E16AS_V5"
+)
+
+type AzureCDMVersionTag struct {
```

Comment:
All exported (CamelCased) types should be documented. There are a couple of more below in the file.